clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -200,7 +200,7 @@ public synchronized boolean containsTopic(String topic) {
      * Updates the cluster metadata. If topic expiry is enabled, expiry time
      * is set for topics if required and expired topics are removed from the metadata.
      */
-    public synchronized void update(Cluster cluster, long now) {
+    public synchronized void update(Cluster cluster, boolean hasUnavailablePartitions, long now) {
```

Comment:
Would it be a bit more general if we passed the unavailable partitions instead of the boolean? It would also be good to add a javadoc explaining what unavailable means. Same for the listener method that takes this variable.